backend/pkg/modules/internal/analytics_core/init.go

```diff
@@ -46,16 +55,42 @@ type AnalyticsCore struct {
 	info                *core.ModuleInfo
 	numWorkers          int
 	proccFuncRegistered map[TopicType][]AnalyticsModuleProccFunc
+	dataFramesRegistry  DataFramesRegistry
 	topics              []TopicType
 }
 
-type ProcFuncDataFrames struct {
-	dataFrames map[int]*interface{}
+type DataFramesRegistry struct {
+	dataFrames map[DataFrameID]DataFrame
+}
+
+func NewDataFramesRegistry(numWorkers int) DataFramesRegistry {
+	return DataFramesRegistry{
+		dataFrames: map[DataFrameID]DataFrame{},
+	}
+}
+
+func (dfr DataFramesRegistry) NewDataFrame(name DataFrameID, shards int) error {
```

Comment:
I think we need here as a parameter size limit of the data frame and optionally TTL